### PR TITLE
Clean up git-series packaging, use OpenSSL 1.1

### PR DESCRIPTION
**Pull request by edef1c**
###### Motivation for this change

This cleans up the git-series packaging to use fewer vendored native libraries, and minimally upgrades some of the Cargo dependencies to allow using recent versions of OpenSSL.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @vmandela

### CODE REVIEW

File: pkgs/development/tools/git-series/default.nix [hunk 1]
```diff
@@ -13,21 +13,22 @@ buildRustPackage rec {
...
-    )
-  '';
+  cargoSha256 = "09y4fvv279cm5sgbpmskascb7rrwqgya5w5wkpz1bgx6pzw3mc7y";
+
+  cargoPatches = [
+    (fetchpatch {
+      url = "https://github.com/Mic92/git-series/commit/3aa30a47d74ebf90b444dccdf8c153f07f119483.patch";
+      sha256 = "06v8br9skvy75kcw2zgbswxyk82sqzc8smkbqpzmivxlc2i9rnh0";
+    })
+    (fetchpatch {
```
Comment (worldofpeace) [inline pkgs/development/tools/git-series/default.nix]:
```suggestion
    # Update cargo.lock allowing OpenSSL 1.1.1 to be used.
    (fetchpatch {
```

**Review by worldofpeace — APPROVED** [review]
LGTM, though upstream really needs to make a release soon. Things like having problems with latest rust and OpenSSL 1.1 isn't going to be good for them. Judging from recent commit activity I hope there will be one.

We could update to an unstable commit, but patching is just really the best way to go.

**Comment by edef1c** [comment]
> Judging from recent commit activity I hope there will be one.
Oh! That's a pleasant surprise. When I wrote this originally (and mailed it out for upstreaming), there hadn't been any activity in 3 years.

**Comment by worldofpeace** [comment]
Hmm, judging from
* https://github.com/git-series/git-series/commit/778536af2b444789067dc167b4a1b51d913d50f1
* https://github.com/git-series/git-series/commit/c3d1a89f21ce33420d7a0634a2ae6c8cb6ada441
* https://github.com/git-series/git-series/commit/dd7e3608993de00e545dd2a8d7ccf7aac0199694
* https://github.com/git-series/git-series/commit/db4dd2c85652f1746bec1915fba4cc9469f7aaca

maybe more advisable to use an unstable version?

**Comment by edef1c** [comment]
Perhaps, but this patch is strictly more conservative than that. I have more confidence in this version than in an unreleased version that upgrades major versions of more complex dependencies.

**Comment by worldofpeace** [comment]
> Perhaps, but this patch is strictly more conservative than that. I have more confidence in this version than in an unreleased version that upgrades major versions of more complex dependencies.

Correct, and you volunteer to maintain this anyway :+1:

**Pull request merged**